utils/resultstable.go

```diff
@@ -258,10 +258,37 @@ func sortVulnerabilityOrViolationRows(rows []formats.VulnerabilityOrViolationRow
 		if rows[i].SeverityNumValue != rows[j].SeverityNumValue {
 			return rows[i].SeverityNumValue > rows[j].SeverityNumValue
 		}
-		return len(rows[i].FixedVersions) > 0 && len(rows[j].FixedVersions) > 0
+		if rows[i].Applicable != rows[j].Applicable {
+			return convertApplicableToScore(rows[i].Applicable) > convertApplicableToScore(rows[j].Applicable)
+		}
+		if ok, compareResult := CompareJfrogResearchInformation(rows[i].JfrogResearchInformation, rows[j].JfrogResearchInformation); ok {
+			return compareResult
+		}
+		return rows[i].IssueId > rows[j].IssueId
 	})
 }
 
+// CompareJfrogResearchInformation compares between two JFrog Research imformation pointers
+// It returns two boolean values: First to validate if comparison is needed and second is the comparison result
+// If both values don't exist return false in first value no need to compare
+// If one of them exist and the other not return true if only the first exist and false if only the second
+// If both exist check if the value is different and compare the values
+// If values are equal return false in the first value
+func CompareJfrogResearchInformation(jfrogResearchInformation1 *formats.JfrogResearchInformation, jfrogResearchInformation2 *formats.JfrogResearchInformation) (bool, bool) {
```

Comment:
you can create a method to get priority of jfrogResearchInformation1 that will handle the nil.
In addition to make this current method return one value, you can use the priority jus like you are comparing at the main func that calls it..